### DISCUSSION TITLE
May I know if "yazi" supports automatic line wrapping in file preview？

### DISCUSSION THREAD

**Discussion by Oyyko** [sxyazi/yazi#265] [4. Ideas]
Sometimes, for instance when viewing HTML files, there can be extremely long single lines in the file. In such cases, the automatic line wrapping feature can be helpful. Additionally, to aid this functionality, the preview feature may benefit from having line numbers.

**Comment by sxyazi** [comment]
I've thinking about this as well. HTML with extremely long lines is generally compressed or obfuscated and is not inherently readable, and it's not clear whether splitting it up and displaying it would make it readable.

Also, whether to add line breaks or horizontal scrolling support is another thing to consider. At the moment, line breaks only apply to textual content such as code, but horizontal scrolling applies to a much wider range of content, such as directories, archive files, etc.

**Reply by sijow** [reply]
Indeed line wrapping would be nice for unstructured text (showing long lines as paragraphs) while horizontal scrolling is probably preferred for structured content. For example I was looking for a horizontal scrolling feature to preview CSV files (with the miller plugin).

Maybe something else to consider is that line wrapping would be easy to do as a plugin I guess while scrolling not so much?

**Comment by mskvsk** [comment]
Just chiming in on the topic.

My MD and TXT files are not word wrapping and I would really love to have this feature.
I do not mind my code to word wrap as well.

So an option for a hard word wrapping would be handy.

**Reply by mskvsk** [reply]
@sxyazi I'd love to have your input on if Yazi not wrapping MD and TXT is a bug or we need to discuss some new approach on word wrapping the preview.  I'm happy to create an issue if it's the former or provide some ideas if it's the latter.

**Comment by sxyazi** [comment]
Done in https://github.com/sxyazi/yazi/pull/1159 🎉
